core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -1202,21 +1192,64 @@ public boolean isIndexTable() throws IOException {
     }
   }
 
+  public List<String> getIndexTableNames(String indexProvider) throws IOException {
+    deserializeIndexMetadata();
+    if (null != indexMetadata) {
+      return indexMetadata.getIndexTables(indexProvider);
+    } else {
+      return new ArrayList<>();
+    }
+  }
+
   public String getIndexInfo() throws IOException {
+    return getIndexInfo(null);
+  }
+
+  public IndexMetadata getIndexMetadata() throws IOException {
+    deserializeIndexMetadata();
+    return indexMetadata;
+  }
+
+  public String getIndexInfo(String indexProvider) throws IOException {
     deserializeIndexMetadata();
     if (null != indexMetadata) {
-      IndexTableInfo[] indexTableInfos =
-          new IndexTableInfo[indexMetadata.getIndexesMap().entrySet().size()];
-      int index = 0;
-      if (!isIndexTable()) {
-        for (Map.Entry<String, List<String>> entry : indexMetadata.getIndexesMap().entrySet()) {
-          indexTableInfos[index] =
-              new IndexTableInfo(getDatabaseName(), entry.getKey(), entry.getValue());
-          index++;
+      if (null != indexProvider) {
+        if (null != indexMetadata.getIndexesMap().get(indexProvider)) {
+          IndexTableInfo[] indexTableInfos =
+              new IndexTableInfo[indexMetadata.getIndexesMap().get(indexProvider).entrySet()
+                  .size()];
+          int index = 0;
+          if (!isIndexTable()) {
```

Comment:
why this check is required ? `if (!isIndexTable()) {`

Reply:
added